src/app/clusters/ota-requestor/GenericOTARequestorDriver.cpp

```diff
@@ -146,14 +147,18 @@ void GenericOTARequestorDriver::UpdateNotFound(UpdateNotFoundReason reason, Syst
         break;
     case UpdateNotFoundReason::kBusy:
         willTryAnotherQuery = true;
-        break;
+        if (mProviderRetryCount <= kMaxBusyProviderRetryCount)
+        {
+            break;
+        }
+        ChipLogProgress(SoftwareUpdate, "Max Busy Provider retries reached. Attempting to get next Provider.");
+        __attribute__((fallthrough)); // fallthrough
```

Comment:
Is there a complaint from CI about this if we just did an intentional fallthrough without this line?

Reply:
The build system is setup with the "-Wimplicit-fallthrough" compiler option, which throws a warning, yes.

Reply:
I see, I didn't see that locally

Reply:
I think we would want to fall through, yes. Imagine the application layer (or periodic query) requesting an OTA. If the first provider remains busy, then I would want the driver to try the next one on the list, and so on, until an OTA either succeeds, or the list has been traversed entirely once. After this we fall back to the 24 hour polling (where this whole FSM repeats).

Reply:
We cleared this up offline. The summary is that since we'll always start from the beginning of the provider list from every new image booted up, we will want to allow iteration through the entire list before giving up and letting the periodic timer kick in